Add the azure openai finetune connector (#756)
ballerina-platform/openapi-connectors: openapi/azure.openai.finetunes/Module.md

## Overview
This is a generated connector from [Azure OpenAI Files API](https://learn.microsoft.com/en-us/rest/api/cognitiveservices/azureopenaistable/files/),

> [Azure OpenAI Fine Tunes API](https://learn.microsoft.com/en-us/rest/api/cognitiveservices/azureopenaistable/fine-tunes/) and

Reviewer: ```suggestion
[Azure OpenAI Fine Tunes API](https://learn.microsoft.com/en-us/rest/api/cognitiveservices/azureopenaistable/fine-tunes/), and
```

Author: Modified module.md file according to the comment

[Azure OpenAI Models API](https://learn.microsoft.com/en-us/rest/api/cognitiveservices/azureopenaistable/models/) OpenAPI specification.

> The Azure OpenAI Files API give access to data files related operations such as delete, get, get content, import, list and upload. The Azure OpenAI Fine Tunes API give access to fine tuning of base models and related operations such as create, cancel, delete, get, get event and list. The Azure OpenAI  Models API give access to deteails about the existing models.

Reviewer: ```suggestion
The Azure OpenAI Files API gives access to data files related operations such as delete, get, get content, import, list, and upload. The Azure OpenAI Fine Tunes API gives access to fine-tuning of base models and related operations such as create, cancel, delete, get, get event, and list. The Azure OpenAI  Models API gives access to details about the existing models.
```

Author: Modified module.md file according to the comment

## Prerequisites
- Create an [Azure](https://azure.microsoft.com/en-us/features/azure-portal/) account.
- Create an [Azure OpenAI resource](https://learn.microsoft.com/en-us/azure/cognitive-services/openai/how-to/create-resource).

> - Obtain the tokens. Refer [this](https://learn.microsoft.com/en-us/azure/cognitive-services/openai/reference#authentication) guide to learn how to generate and use tokens.

Reviewer: ```suggestion
- Obtain the tokens. Refer to [Azure OpenAI Authentication](https://learn.microsoft.com/en-us/azure/cognitive-services/openai/reference#authentication) guide to learn how to generate and use tokens.
```

Author: Modified module.md file according to the comment

## Quickstart

To use the Azure OpenAI Fine Tunes connector in your Ballerina application, update the .bal file as follows:

> ### Step 1: Import connector

Reviewer: ```suggestion
### Step 1: Import the connector
```

Author: Modified module.md file according to the comment

Import the `ballerinax/azure.openai.finetunes` module into the Ballerina project.

```ballerina

> import ballerinax/azure.finetunes;

Reviewer 2: ```suggestion
import ballerinax/azure.openai.finetunes;
```

```

### Step 2: Create a new connector instance

> Create and initialize a `finetunes:Client` with the obtained `apiKey` and a `serviceUrl` from the azure OpenAI resource

Reviewer: ```suggestion
Create and initialize a `finetunes:Client` with the obtained `apiKey` and a `serviceUrl` from the Azure OpenAI resource
```

Author: Modified module.md file according to the comment

```ballerina
final finetunes:Client finetunesClient = check new (
    apiKeyConfig = {auth: {apiKey: apiKey}},
    serviceUrl = serviceUrl
);
```

### Step 3: Invoke connector operation
1. Now you can use the operations available within the connector. 

>**Note:** that they are in the form of remote operations.

> Following is an example on uploading a file and finetuning curie base model:

Reviewer: ```suggestion
Following is an example of uploading a file and finetuning curie base model:
```

Author: Modified module.md file according to the comment

Author: Modified module.md file according to the comment

> Uploading the data file from local machine.

Reviewer: ```suggestion
Uploading the data file from the local machine.
```

Author: Modified module.md file according to the comment
